internal/provider/provider.go

```diff
@@ -224,11 +224,23 @@ func New(version string) func() provider.Provider {
 	}
 }
 
-func (p *NuoDbaasProvider) ConfigValidators(context.Context) []provider.ConfigValidator {
-	return []provider.ConfigValidator{
-		providervalidator.RequiredTogether(
-			path.MatchRoot("user"),
-			path.MatchRoot("password"),
-		),
+func (p *NuoDbaasProvider) ValidateConfig(ctx context.Context, req provider.ValidateConfigRequest, resp *provider.ValidateConfigResponse) {
+	var config NuoDbaasProviderModel
+	if !framework.ReadResource(ctx, &resp.Diagnostics, req.Config.Get, &config) {
+		return
+	}
+
```

Comment:
The `Configure()` function already does validation? Why not put this there so that we report all errors? I suspect breaking this up into two methods would cause errors validated in `Configure()` to not show if there are errors here.

I guess you can move all the validation here, but I don't see how having two methods that have to deserialize the config is better than one, and I'd rather validate the config right before using it as a sanity check in case a lifecycle violation occurs and the `ValidateConfig()` method is not invoked first somehow.

Reply:
The more "correct to the documentation" solution is to move the validation of URL and timeouts into the validate method, but you are right that that would mean that we are parsing the plan and timeouts twice.

It would come down to how many of our users run `terraform validate`. Since I have not heard any of them asking about it, I suspect there are none.

Reply:
I did not realize that `terraform validate` only invokes the `Validate()` function and not the `Configure()` function. In that case, it does make sense to move all validation there, or to some third function that can be invoked by both.

You just have to remember to check `resp.Diagnostics` in the `Configure()` function.